### PR TITLE
Remove configuration settings equals

### PR DESCRIPTION
**Pull request by mssfang**
fixes: #5434

### CODE REVIEW

**Comment by alzimmermsft** [comment]
I'm certain unit tests were using this and they'll need to be updated

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java [hunk 1]
```diff
@@ -412,9 +411,32 @@ static void assertConfigurationEquals(ConfigurationSetting expected, Response<Co
     static void assertConfigurationEquals(ConfigurationSetting expected, ConfigurationSetting actual) {
         if (expected != null && actual != null) {
             actual = cleanResponse(expected, actual);
+        } else if (expected == actual) {
+            return;
+        } else if (expected == null || actual == null) {
+            assertTrue(false);
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: You can do an assert fail here instead with a message.

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java [hunk 2]
```diff
@@ -412,9 +411,32 @@ static void assertConfigurationEquals(ConfigurationSetting expected, Response<Co
...
+        assertEquals(expected.getLastModified(), actual.getLastModified());
+        assertEquals(expected.getContentType(), actual.getContentType());
+
+        final Map<String, String> expectedTags = expected.getTags();
+        final Map<String, String> actualTags = actual.getTags();
+
+        boolean expectedIsNullOrEmpty = ImplUtils.isNullOrEmpty(expectedTags);
+        boolean actualIsNullOrEmpty = ImplUtils.isNullOrEmpty(actualTags);
+
+        if (expectedIsNullOrEmpty) {
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: can you do the following instead of this? It's hard to follow... especially with that assertTRue(true) down there.

```java
assertEquals(expectedIsNullOrEmpty, actualIsNullOrEmpty);
assertEquals(expectedTags, actualTags);
```

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java [hunk 3]
```diff
@@ -488,4 +510,64 @@ static void assertRestException(Throwable exception, Class<? extends HttpRespons
...
+            || !Objects.equals(o1.getETag(), o2.getETag())
+            || !Objects.equals(o1.getLastModified(), o2.getLastModified())
+            || !Objects.equals(o1.isLocked(), o2.isLocked())
+            || !Objects.equals(o1.getContentType(), o2.getContentType())
+            || ImplUtils.isNullOrEmpty(o1.getTags()) != ImplUtils.isNullOrEmpty(o2.getTags())) {
+            return false;
+        }
+
+        if (!ImplUtils.isNullOrEmpty(o1.getTags())) {
+            return Objects.equals(o1.getTags(), o2.getTags());
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: Is this going to perform an object reference comparison rather than the contents?

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java [hunk 4]
```diff
@@ -488,4 +510,64 @@ static void assertRestException(Throwable exception, Class<? extends HttpRespons
...
+        if (settings1 == null || settings2 == null) {
+            return false;
+        }
+
+        if (settings1.size() != settings2.size()) {
+            return false;
+        }
+
+        final int size = settings1.size();
+        for (int i = 0; i < size; i++) {
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: What if the contents aren't ordered in the same way? Will that affect this equality?

Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: It is list, so i think the order matters.

Comment (conniey) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: Does the REST API guarantee an order that the configuration settings are returned in?

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java (hunk 3 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: getTags() returns Map.  equals(Map, Map) compares contents as well. https://howtodoinjava.com/java/collections/hashmap/compare-two-hashmaps/

Comment (conniey) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: Hmm interesting. Could you simplify this logic to:
```java
return Objects.equals(o1.getKey(), o2.getKey())
    && Objects.equals(o1.getLabel(), o2.getLabel())
    && Objects.equals(o1.getValue(), o2.getValue())
    && Objects.equals(o1.getETag(), o2.getETag())
    && Objects.equals(o1.getLastModified(), o2.getLastModified())
    && Objects.equals(o1.isLocked(), o2.isLocked())
    && Objects.equals(o1.getContentType(), o2.getContentType())
    && ImplUtils.isNullOrEmpty(o1.getTags()).equals(ImplUtils.isNullOrEmpty(o2.getTags())))
    && Objects.equals(o1.getTags(), o2.getTags());
```

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java (hunk 4 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java]: Not sure what the REST API guaranteed applying to this method.  But I can explain what I think of. This equal method is a helper function that could be one of many equals() method users want to define how is equality of two settings. It just could be one of them.
